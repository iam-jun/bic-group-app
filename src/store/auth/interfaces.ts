import {CognitoUser, ISignUpResult} from 'amazon-cognito-identity-js';
// Actiontype
export interface ISignIn {
  email: string;
  password: string;
}

export interface ISignUp {
  username: string;
  email: string;
  password: string;
}

export interface IForgotPassword {
  email: string;
  callback: () => void;
}

export interface IForgotPasswordRequest {
  code: string;
  email: string;
  password: string;
}

export interface IUser extends CognitoUser {}

export interface ISignUpResponse extends ISignUpResult {}
