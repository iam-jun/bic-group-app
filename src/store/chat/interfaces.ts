import {IMessage as _GMessage} from 'react-native-gifted-chat';
import {IUser} from '../auth/interfaces';
export interface IReaction {
  type: string;
  count: number;
  reacted?: boolean;
}

export type GMessage = _GMessage;

export type IAttachment = {
  name: string;
  url: string;
  thumnailUrl?: string;
  type: string;
};

export type IMessage = _GMessage & {
  id: string | number;
  user: IUser;
  quoted_message?: IMessage;
  reactions?: IReaction[];
  attachments?: IAttachment[];
};

export type IConversation = {
  name: string;
  description?: string;
  members: IUser[];
  unreadCount: number;
  lastMessage: string;
  updatedAt: string;
};
