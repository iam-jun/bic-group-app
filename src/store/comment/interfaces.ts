import {IUser} from '../auth/interfaces';

export type IComment = {
  id: string;
  content: string;
  createdAt: string;
  updateAt: string;
  user: IUser;
  replyCount: number;
};
