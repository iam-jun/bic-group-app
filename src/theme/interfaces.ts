export interface ITheme extends ReactNativePaper.Theme {
  dark: boolean;
  colors: IThemeColors;
  spacing?: ISpacing;
  dimension?: any;
  shadow?: any;
  fontFamily?: any;
  fonts: any;
  animation: any;
}

export interface IThemeColors extends ReactNativePaper.ThemeColors {
  primary: string;
  primary1?: string;
  primary2?: string;
  primary3?: string;
  primary4?: string;
  primary5?: string;
  primary6?: string;
  primary7?: string;

  background: string;
  bgFocus?: string;
  bgHover?: string;
  bgDisable?: string;
  bgButtonPrimary?: string;
  bgButtonSecondary?: string;
  border?: string;
  placeholder: string;
  surface: string;
  iconTint?: string;
  iconTintReversed?: string;

  link?: string;
  activeLink?: string;
  success?: string;
  error: string;
  warning?: string;

  textPrimary?: string;
  textSecondary?: string;
  textSuccess?: string;
  textDanger?: string;
  textWarning?: string;
  textInfo?: string;
  textDisabled?: string;
  textReversed?: string;

  facebook?: string;
  google?: string;
  apple?: string;
}

export interface ISpacing {
  margin: ISpacingValue;
  padding: ISpacingValue;
  borderRadius: ISpacingValue;
  lineHeight: ISpacingValue;
}

export interface ISpacingValue {
  base?: number;
  tiny?: number;
  small?: number;
  large?: number;
  big?: number;
}
