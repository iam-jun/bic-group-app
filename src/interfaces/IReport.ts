export interface IReportReason {
  data: IReason[];
  loading: boolean;
}

export interface IReason {
  id?: string;
  value?: string;
}

export enum TargetType {
  POST = 'POST',
  ARTICLE = 'ARTICLE',
  COMMENT = 'COMMENT',
  CHILD_COMMENT = 'CHILD_COMMENT',
  MEMBER = 'MEMBER',
}

export enum ReportTo {
  GROUP = 'GROUP',
  COMMUNITY = 'COMMUNITY',
}

export interface IPayloadReportContent {
  dataComment?: {
    parentCommentId?: string;
    postId?: string;
  };

  targetId: string;
  groupIds: string[];
  targetType: TargetType;
  reportTo: ReportTo;
  reasonType: string;
  reason: string;
  attachment?: string;
}

export interface IParamsReportContent {
  targetId: string;
  groupIds: string[];
  targetType: TargetType;
  reportTo: ReportTo;
  reasonType: string;
  reason: string;
  attachment?: string;
}

export interface IPayloadReportMember {
  targetId: string;
  communityId: string;
  reason: string;
}

export interface IParamsReportMember {
  userId: string;
  reason: string;
}

export interface IReportedContents {
  ids: string[];
  loading: boolean;
  refreshing: boolean,
  hasNextPage: boolean,
}

export interface IParamGetReportContent {
  order: 'ASC' | 'DESC';
  offset: number;
  limit: number;
  targetIds?: string[];
  targetType?: TargetType;
}

export interface IReportDetail {
  total: number;
  reasonType: string;
  description: string;
}
