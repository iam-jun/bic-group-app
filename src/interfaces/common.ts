import {languages} from '~/configs';
import {IconType} from '~/resources/icons';
import {StreamClient} from 'getstream';
import {ToastMessageProps} from '~/beinComponents/ToastMessage/NormalToastMessage';

export interface IObject<T> {
  [key: string]: T;
}

export interface ILanguage {
  code: keyof typeof languages;
  title: string;
  icon: IconType;
}

export interface ISetting {
  type: string;
  title: string;
  icon: IconType;
}

export interface IGetStreamDispatch {
  streamClient: StreamClient;
  userId: string;
}

export interface IRightMenu {
  type: string;
  title: string;
}

export interface IFilePicked {
  name: string;
  filename: string;
  type: string;
  mime: string;
  size: number;
  width: number;
  height: number;
  //app
  uri: string;
  //web
  base64?: string;
  [x: string]: any;
}

export interface IResponseData {
  code: number;
  data: any;
  meta: any;
}

export interface IToastMessage {
  content: string;
  props: ToastMessageProps;
  duration?: number;
}

export interface IPayloadShowModal {
  isOpen: boolean;
  ContentComponent: any;
}
