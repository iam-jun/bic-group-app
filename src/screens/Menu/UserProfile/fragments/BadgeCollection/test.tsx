import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
