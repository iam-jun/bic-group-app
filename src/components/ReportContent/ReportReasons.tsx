import React from 'react';
import { StyleSheet, ScrollView } from 'react-native';

import { Radio } from '~/baseComponents';
import { spacing } from '~/theme';
import useReportContentStore from './store';
import { IReason, TargetType } from '~/interfaces/IReport';

interface ReportReasonsProps {
  reasonState: IReason;
  targetType: TargetType;
  setReasonState: (item: IReason) => void;
}

const ReportReasons: React.FC<ReportReasonsProps> = ({
  reasonState,
  targetType,
  setReasonState,
}) => {
  const { reportReasons, memberReportReasons } = useReportContentStore((state) => state);
  const reasons = targetType === TargetType.MEMBER ? memberReportReasons.data : reportReasons.data;

  const renderReasons = () => (
    reasons?.map((item) => (
      <Radio
        key={`report_content${item.id}`}
        useI18n
        label={item.value}
        isChecked={reasonState?.id === item.id}
        onPress={() => setReasonState(item)}
        style={styles.radioReason}
      />
    ))
  );

  return (
    <ScrollView style={styles.container}>
      {renderReasons()}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: spacing.margin.large,
  },
  radioReason: {
    marginBottom: spacing.margin.base,
  },
});

export default ReportReasons;
