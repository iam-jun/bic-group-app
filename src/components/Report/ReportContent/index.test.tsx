import React from 'react';
import { renderWithRedux } from '~/test/testUtils';
import ReportReasons from '.';
import { TargetType, IPayloadReportContent, ReportTo } from '~/interfaces/IReport';

describe('ReportContent component', () => {
  const baseProps = {
    targetId: 'ec383429-e870-4ad3-9322-9857b4bb4769',
    groupIds: [
      'b01fb58e-9299-4a0e-a55f-9839293fb42a',
    ],
    targetType: TargetType.COMMENT,
    reportTo: ReportTo.COMMUNITY,
  } as IPayloadReportContent;

  it('renders correctly', () => {
    const rendered = renderWithRedux(
      <ReportReasons
        {...baseProps}
      />,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
