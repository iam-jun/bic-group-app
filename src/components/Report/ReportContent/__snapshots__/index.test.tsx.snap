// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportContent component renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "marginRight": 8,
            },
            Object {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#181C32",
              "fontFamily": "BeVietnamPro-SemiBold",
              "fontSize": 17,
            },
            Object {
              "color": "#181C32",
            },
            undefined,
          ]
        }
      >
        Report Content to Community Admin
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 466.9,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#BABEC9"
      size="small"
    />
  </View>
</View>
`;
